src/posts/2020-07-27-unsucking-your-project-onboarding.md

```diff
@@ -0,0 +1,789 @@
+  changed in Javascript landscape since 37s ago. Now they have to install the
+  backend, frontend, all the tooling, etc etc etc. Stop this madness. The
+  programming languages and frameworks should be accessories to what you're doing.
+  They're tools you use and making use of them should be as simple as possible. At
+  Subvisual we have a system to handle this.
+---
+
+So you have a new developer joining your project. And, as usual, everything
+changed in Javascript landscape since 37s ago. Everyone is already on a shiny
+new framework. Of course, you, being an amazing developer, are ahead of the
+landscape for at least 112s! You're already 3 frameworks and 7 minor versions
+ahead.
+
+However, the new developer isn't. And now they have to install the new backend
+language version, let's say Elixir, which by the way needs this really specific
+OTP 22 version, oh and let's not forget the frontend. Yeah, we're going to need
+a new node version which might mean having to import the team keyring, the whole
+Webpack thingy. Wait, you used `npm`? Nope, not gonna happen, we use `yarn`.
+Yeah, run that again and delete this file. Is it done? Ok, you can run the app!
+
+It's not working? Oh yeah, the environment variables. We need them to start the
+app in development sometimes. Well, I'll send them to you through Slack even
+though Slack isn't encrypted.
+
+Ah, let's not forget setting up the database, migrating and seeding it. I guess
+that's all? Oh, how could we have forgotten... To use Phoenix you need to do the
+archive install. You're gonna need to do that as well.
+
+Well, this took all week. See you Monday!
+
+.
+
+.
+
+.
+
+.
+
+.
+
+Sound familiar? In **every** (and I mean every) project Subvisual does, we use
+`bin/` scripts. And not just to set up the project. **The programming languages
+and frameworks should be accessories to what you're doing. They're tools you use
+and making use of them should be as simple as possible.** In every project we
+have a setup to do, a server to run and linters to check. So we abstract those
+steps with `bin/setup`, `bin/server` and `bin/lint`.
+
+This is particularly important for open source projects where you want people to
+join in and help with as little friction as possible.
+
+These first three scripts are so common, I have `s`, `se` and `l` aliased to
+`bin/{server,setup,lint}` just to avoid having to type all those extra letters.
+
+We don't stop there. **If anything requires you to memorise a command, we
+automate it.** This allows us to onboard people into the project by doing a
+simple `git clone` and `bin/setup`. **Time to first contribution is a couple of
+hours instead of 1+ days**.
+
+Let's get right into the basic scripts, the advanced scripts and the insane ones.
+
+## Into the basics
+
+As I mentioned, there are three essential scripts: `bin/server`, `bin/setup` and
+`bin/lint`. The first runs the development server. The second installs
```

Comment:
again, `bin/test` missing here. might not be as common, but then again, it's usually a counterpart to `bin/lint`